### DISCUSSION TITLE
Your update DoS:ed my webhooks

### DISCUSSION THREAD

**Discussion by haf** [swellstores/community#205] [Help] [answered]
At the same time you had the latest massive outage, when you came back online, I believe your system spammed my Vercel webhooks to oblivion, causing my account to get banned:

<img width="1111" alt="image" src="https://user-images.githubusercontent.com/193115/230663681-4afa06df-612f-4685-a912-70ac2b8c427e.png">

**Comment by ericingram** [comment]
Not good! Do you need any help fixing it?

**Reply by haf** [reply]
I would love to get some support on how to keep my webhooks alive. I had to move to a self-hosted server because of this, and this server is only turned on 16 hours a day, which causes your webhooks to become disabled.

Additionally, one of the product-changed webhooks keep firing for a non-existent product, which I believe is an extra reason that this webhook is disabled after a while?

Perhaps you can enable in-store support for haf-se and I can ping you in the internal chat instead of writing publicly?

This is my stdout from my service when called from swell:

```json
{"ok":false,"message":"Product not found with id=63c2e44fdb51c50012431ac9","body":{"model":"products","type":"product.updated","data":{"id":"63c2e44fdb51c50012431ac9","options":[null,{"attribute_id":"frames","id":"63c2e452e832970013bdd077"}]},"date_created":"2023-01-14T17:20:19.297Z","date_updated":"2023-04-12T14:06:06.934Z","webhooks_pending":1,"id":"63c2e453e832970013bdd086"}}
```

**Reply by ericingram** [reply, marked as answer]
@haf The simplest fix for the missing product issue is to return a status `200` if the product is not found. Essentially, any scenario that you don't want your webhook to handle, can stop further requests by returning 200.

Unfortunately I don't think we have a solution for the part-time server situation.

Re: support, this is the best place to get it for technical needs. Actually, more of our engineers participate in this forum than in the internal chat, although the priority is lower.

FYI, we are looking at introducing a low-cost paid plan that includes chat/email support this year.

**Comment by osseonews** [comment]
This is exactly what happened to us on Netlify. Netlify had our usage yesterday at 9 million requests in a few hours and over 19,000 hours of function run time.  We maybe get 100 visitors a day to our website max. Not only was our site down, we were billed $800 in a few hours for this usage.  Outrageous. Will Swell compensate us for the damage they cost us with their utter negligence?

**Reply by ericingram** [reply]
@osseonews I would suggest talking to our support team and sharing details of those 9M requests. It would be abnormal, but if there was a runaway process in this case we'd like to investigate further

**Reply by osseonews** [reply]
Of course, this is an abnormal is the context of normal functioning API endpoint. There are few companies in the world that have real usage of 9 million requests in a few hours. We certainly are not one of them.  But, Swell's API was not functioning normally for nearly 12+ hours on April 5 to April 6. We were in touch with support for hours during the outage and they knew exactly what was going on. Unfortunately, in addition to doing zero QA prior to the major upgrade, you had nobody monitoring servers during the outage, and nobody initially on call for hours who understand how to fix the issues with the API.  Fortunately, at some point, somebody who understood how to fix the issue, applied a fix and the "DDOS" style attack ended. Do you seriously believe a customer is responsible for paying for overage caused by Swell's malfunctioning API which did the equivalent of a DDOS attack on customer servers for hours?

**Reply by ericingram** [reply]
If you can share specific details about what happened with our support team we'll look into it. It should show up in logs but so far the team hasn't been able to confirm what you're describing.

**Reply by osseonews** [reply]
Nonsense. You know very well what happened. Logs? Your own status page showed a problem. Your API was malfunctioning. It  DDOSed our website and others. The proper response is to take responsibility and compensate customers in some way for the losses caused the negligence of Swell during this major upgrade without any QA, or anyone on call to monitor it.
